.github/workflows/release.yml

```diff
@@ -347,36 +347,20 @@ jobs:
     if: needs.preflight.outputs.skip-publishing == 'false' || github.event.inputs.dry-run == 'true'
 
     steps:
-      - name: Check out ${{ github.repository }}
-        uses: actions/checkout@v3
-
-      - name: Install Devolutions Toolbox
-        uses: devolutions/actions/toolbox-install@master
-        with:
-          token: ${{ secrets.DEVOLUTIONSBOT_TOKEN }}
+      ## Download back the artifacts produced by the other jobs
 
-      - name: Download cargo-cyclonedx from OneDrive
-        uses: devolutions/actions/download-onedrive@master
+      - name: Download artifacts (action)
+        if: needs.preflight.outputs.dl-strategy == 'action'
+        uses: actions/download-artifact@v3
         with:
-          files: /Binaries/cargo-cyclonedx/linux/cargo-cyclonedx
-          recursive: 'false'
-          destination_path: /home/runner/.cargo/bin/
-          remote: dev
-          client_id: ${{ secrets.ONEDRIVE_AUTOMATION_CLIENT_ID }}
-          client_secret: ${{ secrets.ONEDRIVE_AUTOMATION_CLIENT_SECRET }}
-
-      - name: Set execute permission on cargo-cyclonedx
-        run: |
-          chmod +x /home/runner/.cargo/bin/cargo-cyclonedx
-
-      - name: Generate SBOM
-        uses: devolutions/actions/cyclonedx-rust-generate@master
+          name: devolutions-gateway
 
-      - name: Save SBOM
-        uses: actions/upload-artifact@v3
-        with:
-          name: bom.xml
-          path: bom.xml
+      - name: Download artifacts (cli)
+        if: needs.preflight.outputs.dl-strategy == 'cli'
+        shell: pwsh
+        env:
+          GITHUB_TOKEN: ${{ secrets.DEVOLUTIONSBOT_TOKEN }}
+        run: gh run download ${{ needs.preflight.outputs.run }} -n devolutions-gateway --repo $Env:GITHUB_REPOSITORY
 
       - name: Upload SBOM
```

Comment:
You should check if dry-run == false here, otherwise you'll get BOM uploads for dry runs (where we run the workflow to check it but without actually publishing anything)